lib/pavilion/plugins/commands/show.py

```diff
@@ -385,65 +412,117 @@ def _functions_cmd(self, _, args):
         """List all known host files."""
 
-        hosts = []
-        col_names = ['Name']
-        if args.verbose:
-            col_names.append('Path')
-        for conf_dir in pav_cfg.config_dirs:
-            path = conf_dir / 'hosts'
+        if args.config is None:
+            hosts = []
+            col_names = ['Name']
+            if args.verbose:
+                col_names.append('Path')
+            if args.vars:
+                col_names.append('Variables')
+            for conf_dir in pav_cfg.config_dirs:
+                path = conf_dir / 'hosts'
 
-            if not (path.exists() and path.is_dir()):
-                continue
+                if not (path.exists() and path.is_dir()):
+                    continue
 
-            for file in os.listdir(path.as_posix()):
+                for file in os.listdir(path.as_posix()):
+
+                    file = path / file
+                    if file.suffix == '.yaml' and file.is_file():
+                        host_id = file.stem
+                        host_path = file
+                        with file.open() as config_file:
+                            config_data = yc_yaml.load(config_file)
+                        try:
+                            host_vars = list(config_data['variables']
+                                             .keys())
+                        except (KeyError, TypeError) as err:
+                            host_vars = []
+
+                        hosts.append({
+                            'Name': host_id,
+                            'Path': host_path,
+                            'Variables': host_vars
+                        })
 
-                file = path / file
-                if file.suffix == '.yaml' and file.is_file():
-                    host_id = file.stem
-                    host_path = file
-                    hosts.append({
-                        'Name': host_id,
-                        'Path': host_path
-                    })
+            output.draw_table(
+                self.outfile,
+                field_info={},
+                fields=col_names,
+                rows=hosts
+            )
+
+        else:
+
+            host = args.config
+            config_data, file = self.get_config(pav_cfg, host, 'hosts')
+            if config_data is not None:
+                output.fprint("Host config for {} found at:{}".format(host,
+                                                                       str(file)),
+                               file=self.outfile)
+                output.fprint(pprint.pformat(config_data, compact=True),
+                              file=self.outfile)
+            else:
+                output.fprint("No host config found for {}.".format(mode))
+                return errno.EINVAL
 
-        output.draw_table(
-            self.outfile,
-            field_info={},
-            fields=col_names,
-            rows=hosts
-        )
 
     @show_cmd('mode')
```

Comment:
Wait a second, isn't 99.9% of this code the same as the above?

You should make this stuff generic.